File: src/main/java/com/iprzd/zshop/entity/PayCard.java
package com.iprzd.zshop.entity;

import lombok.*;

import java.io.Serializable;
import java.util.Date;

@Getter
@Setter
@NoArgsConstructor
@EqualsAndHashCode
@ToString
public class PayCard {

    protected long id;
    protected int isValid;
    protected long accountId;
    protected String account;
    protected String note;
    protected Date createDate;
    protected int expiryMonth;
    protected String number;
    protected String verifyCode;
    protected int denomination;
    protected int amount;
    protected int isSoldOut;
    protected String customer;
    protected String customerPhone;
    protected int isUsed;
    protected String user;
    protected String userId;
    protected Date useDate;
}

File: admin-ui/src/app/response/pay-card.ts

export interface PayCard {
    id?: number;
    isValid: number;
    accountId: number;
    account: string;
    note?: string;
    createDate: Date;
    expiryMonth: number;
    number?: number;
    verifyCode?: number;
    denomination: number;
    amount: number;
    isSoldOut?: number;
    customer?: string;
    customerPhone?: string;
    isUsed?: number;
    user?: string;
    userId?: number;
    useDate?: Date;
}

export interface PayCardCreateModel {
    id?: number;
    count: number;
    note: string;
    denomination: number;
    expiryMonth: number;
    amount: number;
    customer: string;
    customerPhone: string;
}
